src/lambda_handler.py

```diff
@@ -1,22 +1,152 @@
+        response["Credentials"]["SecretAccessKey"],
+        response["Credentials"]["SessionToken"],
+    )
+
+
+def create_assumed_aws_client(
+    aws_service: str, role_arn: str, session_name: str
+) -> boto3.client:
+    """Assume the given role and return an AWS client for the given service using that role."""
+    role_credentials: Tuple[str, str, str] = assume_role(role_arn, session_name)
+
+    return boto3.client(
+        aws_service,
+        aws_access_key_id=role_credentials[0],
+        aws_secret_access_key=role_credentials[1],
+        aws_session_token=role_credentials[2],
+    )
+
+
+def create_assumed_aws_resource(
+    aws_service: str, region: str, role_arn: str, session_name: str
+) -> boto3.resource:
+    """Assume the given role and return an AWS resource object for the given service using that role."""
+    role_credentials: Tuple[str, str, str] = assume_role(role_arn, session_name)
+
+    return boto3.resource(
+        aws_service,
+        region_name=region,
+        aws_access_key_id=role_credentials[0],
+        aws_secret_access_key=role_credentials[1],
+        aws_session_token=role_credentials[2],
+    )
+
+
+def convert_tags(aws_resource: boto3.resource) -> Dict[str, str]:
+    """Convert resource tags from an AWS dictionary into a Python dictionary."""
+    try:
+        tags: Dict[str, str] = {x["Key"]: x["Value"] for x in aws_resource.tags}
+    except TypeError:
+        # This happens if there are no tags associated with the resource
+        tags = {}
+    return tags
+
+
+def get_ec2_ips(
+    ec2: boto3.resource, application_tag_name: str, publish_egress_tag_name: str
+) -> Iterator[Tuple[str, str]]:
+    """Create a set of public EC2 IPs.
+
+    Yields (application tag value, public_ip) tuples.
+    """
+    # Get list of running EC2 instances
+    instances: boto3.resources.collection = ec2.instances.filter(
+        Filters=[{"Name": "instance-state-name", "Values": ["running"]}]
+    )
+    # Get list of elastic IPs in all VPCs
+    vpc_addresses: boto3.resources.collection = ec2.vpc_addresses.all()
+
+    for instance in instances:
+        # If the instance doesn't have a public IP, we can skip it
+        if instance.public_ip_address is None:
+            continue
+
+        # Convert instance tags from an AWS dictionary into a Python dictionary
+        tags = convert_tags(instance)
+
+        # If the publish egress tag doesn't exist or isn't set to True, skip it
+        if tags.get(publish_egress_tag_name, str(False)) != str(True):
+            continue
+        # Send back a tuple associating the public IP to an application.
+        # If application is unset, return "", so that the IP can be included
+        # in a list of all IPs if desired (e.g. using app_regex=".*").
+        yield (tags.get(application_tag_name, ""), instance.public_ip_address)
+
+    for vpc_address in vpc_addresses:
+        # Convert elastic IP tags from AWS dictionary into a Python dictionary
```

Comment:
Consistent commenting.

```suggestion
        # Convert elastic IP tags from an AWS dictionary into a Python dictionary
```